app/src/main/java/io/appium/settings/LocationTracker.java

```diff
@@ -190,29 +204,48 @@ private void stopLocationUpdatesWithoutPlayServices() {
             return;
         }
 
-        Log.i(TAG, "Stopping Android location provider");
+        Log.d(TAG, "Stopping Android location provider");
         mLocationManager.removeUpdates(this);
         mLocationManager = null;
     }
-    
+
+    private boolean isPlayServicesConnected() {
+        return mGoogleApiClient != null && mGoogleApiClient.isConnected();
+    }
+
+    private boolean isLocationManagerConnected() {
+        return mLocationManager != null && mLocationProvider != null;
+    }
+
     @Nullable
-    public Location getLocation() {
+    public synchronized Location getLocation(Context context) {
         if (mLocation == null) {
-            if (mGoogleApiClient != null) {
-                if (mGoogleApiClient.isConnected()) {
-                    try {
-                        //noinspection deprecation
-                        mLocation = LocationServices.FusedLocationApi.getLastLocation(mGoogleApiClient);
-                    } catch (SecurityException e) {
-                        Log.e(TAG, "Appium Settings has no access to location permission");
-                    }
+            // Make sure the service has been started
+            start(context, GOOGLE_API_CONNECT_TIMEOUT, false);
+
+            if (isPlayServicesConnected()) {
+                // Make sure the fallback is removed after play services connection succeeds
+                if (isLocationManagerConnected()) {
+                    stopLocationUpdatesWithoutPlayServices();
                 }
-            } else if (mLocationManager != null && locationProvider != null) {
                 try {
-                    mLocation = mLocationManager.getLastKnownLocation(locationProvider);
+                    //noinspection deprecation
+                    mLocation = LocationServices.FusedLocationApi.getLastLocation(mGoogleApiClient);
                 } catch (SecurityException e) {
-                    Log.e(TAG, String.format("Appium Settings has no access to %s location permission",
-                            locationProvider));
+                    Log.e(TAG, "Appium Settings has no access to location permission", e);
+                }
+            } else if (isLocationManagerConnected() || mGoogleApiClient != null) {
+                // Try fallback to the default provider if Google API is available but is still not connected
+                if (mGoogleApiClient != null && !isLocationManagerConnected()) {
```

Comment:
`mLocationManager` is null if `mGoogleApiClient` was initialized. `startLocationUpdatesWithoutPlayServices()` throws an NPE in that case

Reply:
fixed